### PR TITLE
Move All the Gyb Support for Syntax out of Swift

### PR DESCRIPTION
**Pull request by CodaFi**
Make libSyntax depend on swift-syntax: the new home for all
of this infrastructure. This greatly simplifies the addition and
amending of syntax nodes as only the swift-syntax paired with a
swift checkout will need to be changed. This is in contrast to
the existing build flow where a paired PR to both repos must be
made to change anything here.

Note that a paired PR may still be required if the legacy parser
needs to be adjusted in response to syntax nodes changing, but I
anticipate this to be a much more infrequent event now that
the C++ end of libSyntax is deprecated.

### CODE REVIEW

**Comment by CodaFi** [comment]
Pairs well with apple/swift-syntax#663

File: CMakeLists.txt
Comment (etcwilde) [inline CMakeLists.txt:627]: We should probably stick to a more conventional directory naming scheme.
Something like `SWIFT_SYNTAX_SOURCE_DIR`.

Comment (compnerd) [reply CMakeLists.txt:627]: I think that the name actually works well for the current state of things - all the external repositories use this current naming scheme.

**Comment by compnerd** [comment]
This is going to require a change to compnerd/swift-build to ensure that we do not break the Windows CI there.  Please make sure that we synchronise the changes when this is merged.

File: CMakeLists.txt
Comment (etcwilde) [reply CMakeLists.txt:627]: This is true. Diverges from LLVM, but I guess that's common around these parts.

**Comment by CodaFi** [comment]
apple/swift-syntax#663

@swift-ci test

**Comment by compnerd** [comment]
Do we need to worry about the CI build scripts for Windows?

**Comment by CodaFi** [comment]
apple/swift-syntax#663

@swift-ci test

**Comment by CodaFi** [comment]
https://github.com/apple/llvm-project/pull/5236 will unblock the windows bot. I'll make sure it's blue before I merge.

**Comment by CodaFi** [comment]
apple/swift-syntax#663

@swift-ci test

**Comment by CodaFi** [comment]
https://github.com/apple/swift-syntax/pull/663

@swift-ci test

**Review by stevapple — CHANGES_REQUESTED** [review]
To support this change into Windows CI, we also need to remove line 125 in `build-windows.bat`, and make identical changes to `build-windows-toolchain.bat`. Both files are used on Swift CI.

**Comment by CodaFi** [comment]
https://github.com/apple/swift-syntax/pull/663

@swift-ci test windows

**Comment by CodaFi** [comment]
whoa

**Comment by CodaFi** [comment]
apple/swift-syntax#663

@swift-ci test

**Comment by CodaFi** [comment]
https://github.com/apple/swift-syntax/pull/663

@swift-ci test

**Comment by CodaFi** [comment]
https://github.com/apple/swift-syntax/pull/663

@swift-ci test

**Pull request merged**
